Whiskey/Functions/Import-WhiskeyPowerShellModule.ps1

```diff
@@ -41,33 +41,28 @@ function Import-WhiskeyPowerShellModule
         Get-Module -Name $Name | Remove-Module -Force -WhatIf:$false
     }
 
-    $searchPaths = & {
-        Join-Path -Path $BuildRoot -ChildPath $powerShellModulesDirectoryName
-    }
+    $relativePSModulesRoot = Resolve-Path -Path $PSModulesRoot -Relative -ErrorAction Ignore
 
     foreach( $moduleName in $Name )
     {
-        foreach( $searchDir in  $searchPaths )
+        $moduleDir = Join-Path -Path $PSModulesRoot -ChildPath $moduleName
+        if( (Test-Path -Path $moduleDir -PathType Container) )
         {
-            $moduleDir = Join-Path -Path $searchDir -ChildPath $moduleName
-            if( (Test-Path -Path $moduleDir -PathType Container) )
+            Write-WhiskeyDebug -Message ('PSModuleAutoLoadingPreference = "{0}"' -f $PSModuleAutoLoadingPreference)
+            Write-WhiskeyVerbose -Message ('Importing PowerShell module "{0}" from "{1}".' -f $moduleName,$relativePSModulesRoot)
+            $numErrorsBefore = $Global:Error.Count
+            & {
+                $VerbosePreference = 'SilentlyContinue'
+                Import-Module -Name $moduleDir -Global -Force -ErrorAction Stop -Verbose:$false
+            } 4> $null
+            # Some modules (...cough...PowerShellGet...cough...) write silent errors during import. This causes our tests
+            # to fail. I know this is a little extreme.
+            $numErrorsAfter = $Global:Error.Count - $numErrorsBefore

```

Comment:
Clear global errors prior?

Reply:
I’ll play around with this approach.

Reply:
Looks like that works:

```powershell
$Global:Error.Clear()

for( $idx = 0 ; $idx -lt 10 ; $idx++ )
{
    Write-Error $idx -ErrorAction SilentlyContinue
}

$originalErrors = $Global:Error.Clone()

$Global:Error.Clear()

$Global:Error.AddRange($originalErrors)

$Global:Error
```

So, we could move to this pattern:

```powershell
$originalErrors = $Global:Error.Clone()
$Global:Error.Clear()
try
{
    # Do stuff that might write errors we want to clear.
}
finally
{
    $Global:Error.Clear()
    $Global:Error.AddRange($originalErrors)
}
```